lib/cartopy/mpl/geoaxes.py

```diff
@@ -731,6 +768,19 @@ def set_extent(self, extents, crs=None):
             if boundary.equals(domain_in_crs):
                 projected = boundary
 
+        if clip:
+            n = 100
+            x = np.linspace(x1, x2, n)
+            y = np.linspace(y1, y2, n)
+            xx = np.concatenate((x, [x[-1]]*n, x[::-1], [x[0]]*n))
+            yy = np.concatenate(([y[0]]*n, y, [y[-1]]*n, y[::-1]))
```

Comment:
Can you please explain what `xx` and `yy` are, and add a comment to the same effect?  There's a lot of information to unpack in those two lines and it's making my eyes go funny taking it apart to work out what these variables are and what they represent.

Reply:
Fixed